schema.graphql

```diff
@@ -592,6 +620,215 @@ type AuthzMsgExec @entity {
+  claimedAmount: BigInt
+  claimedDenom: String
+  transaction: Transaction
+  block: Block!
+}
+
+enum ProofRequirementReason {
+  NOT_REQUIRED
+  PROBABILISTIC
+  THRESHOLD
+}
+
+
+type EventClaimSettled @entity {
+  id: ID!
+  supplier: Supplier!
+  application: Application!
+  service: Service!
+  sessionId: String!
+  sessionStartHeight: BigInt!
+  sessionEndHeight: BigInt!
+  rootHash: String
+  proofRequirement: ProofRequirementReason
+  numRelays: BigInt
+  numClaimedComputedUnits: BigInt
+  numEstimatedComputedUnits: BigInt
+  claimedAmount: BigInt
+  claimedDenom: String
+  transaction: Transaction
+  block: Block!
+}
+
+type Relay @entity {
+  id: ID!
+  supplier: Supplier!
+  application: Application!
+  service: Service!
+  sessionId: String!
+  sessionStartHeight: BigInt!
+  sessionEndHeight: BigInt!
+  rootHash: String
+  closestMerkleProof: String
+  numRelays: BigInt
+  numClaimedComputedUnits: BigInt
+  numEstimatedComputedUnits: BigInt
+  claimedAmount: BigInt
+  claimedDenom: String
+  msgCreateClaim: MsgCreateClaim
+  eventClaimCreated: EventClaimCreated
+  eventsClaimUpdated: [EventClaimUpdated] @derivedFrom(field: "relay")
+  eventClaimExpired: EventClaimExpired
+  eventClaimSettled: EventClaimSettled
+  eventProofSubmitted: EventProofSubmitted
+  msgSubmitProof: MsgSubmitProof
+  eventsProofUpdated: [EventProofUpdated] @derivedFrom(field: "relay")
+  # 0 = PENDING, 1 = SUCCESSFUL, 2 = FAILED
+  status: Int!
+  failedCodeSpace: String
+  requiredProof: Boolean
+  # this must be a value of the enums ClaimErrors or ProofErrors. We are using Int because we can't define enums with numbers.
+  failedCode: Int
+}
+
+type EventProofSubmitted @entity {
+  id: ID!
+  supplier: Supplier!
+  application: Application!
+  service: Service!
+  sessionId: String!
+  sessionStartHeight: BigInt!
+  sessionEndHeight: BigInt!
+  rootHash: String
+  closestMerkleProof: String
+  numRelays: BigInt
+  numClaimedComputedUnits: BigInt
+  numEstimatedComputedUnits: BigInt
+  claimedAmount: BigInt
+  claimedDenom: String
+  transaction: Transaction
+  block: Block!
+}
+
+type EventProofUpdated @entity {
+  id: ID!
+  supplier: Supplier!
+  application: Application!
+  service: Service!
+  sessionId: String!
+  sessionStartHeight: BigInt!
+  sessionEndHeight: BigInt!
+  rootHash: String
+  closestMerkleProof: String
+  numRelays: BigInt
+  numClaimedComputedUnits: BigInt
+  numEstimatedComputedUnits: BigInt
+  claimedAmount: BigInt
+  claimedDenom: String
+  transaction: Transaction
+  block: Block!
+  relay: Relay
+}
+
+type RelayByBlockAndService @entity {
+  id: ID!
+  relays: BigInt!
+  computedUnits: BigInt!
+  amount: Int!
+  block: Block!
+  service: Service!
+}
+
+type StakedSuppliersByBlockAndService @entity {
+  id: ID!
+  tokens: BigInt!
+  amount: Int!
+  block: Block!
+  service: Service!
+}
+
+type StakedAppsByBlockAndService @entity {
+  id: ID!
+  tokens: BigInt!
+  amount: Int!
+  block: Block!
+  service: Service!
+}
```

Comment:
#PUC what these entities are for - they seem like they could be join tables but they have additional data on them. Can we not use existing relations to facilitate these kinds of queries?

Reply:
Those entities are being used to save summarized data by block and service.